test/box-luatest/meta-temporary_spaces_test.lua

```diff
@@ -0,0 +1,440 @@
+local fio = require 'fio'
+local xlog = require 'xlog'
+local server = require 'luatest.server'
+local cluster = require 'luatest.replica_set'
+local t = require 'luatest'
+local g = t.group()
+local _ = nil
+
+g.before_all = function()
+    g.server = server:new({alias = 'master'})
+end
+
+g.after_all = function()
+    _ = g.server.process and g.server:stop()
+    _ = g.cluster and g.cluster:drop()
+end
+
+g.before_each(function()
+    if not g.server.process then
+        g.server:start()
+    end
+end)
+
+g.after_each(function()
+    _ = g.server.process and g.server:exec(function()
+        box.cfg { read_only = false }
+        for _, s in pairs({'test', 'temp', 'metatemp'}) do
+            local _ = box.space[s] and box.space[s]:drop()
+        end
+    end)
+end)
+
+-- check basic invariants of meta-temporary spaces
+g.test_meta_temporary_create = function()
+    g.server:exec(function()
+        local t = require 'luatest'
+        local _space = box.space._space
+
+        local s = box.schema.space.create('metatemp', { temporary = 'meta' })
+        t.assert_equals(s.temporary, 'meta')
+
+        local s = box.schema.space.create('metatemp2', { temporary = 'meta',
+                                                         id = 9999 })
+        t.assert_equals(s.temporary, 'meta')
+        s:drop()
+
+        local s = box.schema.space.create('temp', { temporary = true })
+        t.assert_equals(s.temporary, true)
+
+        local s = box.schema.space.create('r1', { temporary = false })
+        t.assert_equals(s.temporary, false)
+        s:drop()
+
+        local s = box.schema.space.create('r2', { temporary = nil })
```

Comment:
You declared `local s` multiple times. Lets do it just once.

Reply:
done